### PR TITLE
Add feature id support

### LINKED ISSUE #59: Preserve feature identifiers
`convert()` [includes](https://github.com/mapbox/geojson-vt-cpp/blob/6ff68fa154b0063125ae9a244e692812eccf3152/include/mapbox/geojsonvt/convert.hpp#L101) the geometry and properties of each feature in the given feature collection, but it throws away the feature’s ID. This means any filter in mapbox-gl-native that depends on the existence of an `$id` property in a GeoJSON source will behave incorrectly.

/ref mapbox/vector-tile-js#46, mapbox/mapbox-gl-native#5708
/cc @jfirebaugh @ivovandongen @cammace

### PR DESCRIPTION
**Pull request by jfirebaugh**
Fixes #59
Ports https://github.com/mapbox/geojson-vt/commit/9ec3ea80fb902cd01c0eaa0a45ca5ebcfef868f0

### CODE REVIEW

File: include/mapbox/geojsonvt/tile.hpp [hunk 1]
```diff
@@ -94,43 +95,43 @@ class InternalTile {
...
                 // `this->` is a workaround for https://gcc.gnu.org/bugzilla/show_bug.cgi?id=61636
-                this->addFeature(g, props);
+                this->addFeature(g, props, id);
             });
         }
     }
 
     template <class T>
-    void addFeature(const T& multi, const property_map& props) {
+    void addFeature(const T& multi, const property_map& props, const std::experimental::optional<identifier>& id) {
```
Comment (mourner) [inline include/mapbox/geojsonvt/tile.hpp]: Maybe alias `std::experimental::optional<identifier>` to `optional_identifier`, which is repeated a lot?

**Comment by jfirebaugh** [comment]
Build is failing due to linker errors that stem from this difference:

```
~/Development/geojson-vt-cpp $ .mason/mason ldflags glfw 3.1.2
-framework Cocoa -framework OpenGL -framework IOKit -framework CoreFoundation -framework CoreVideo
~/Development/geojson-vt-cpp $ .mason/mason ldflags glfw 3.2.1
-framework Cocoa -framework IOKit -framework CoreFoundation -framework CoreVideo
```

@springmeyer @kkaefer Do you know why `-framework OpenGL` is not appearing for glfw 3.2.1?

**Comment by kkaefer** [comment]
@jfirebaugh yeah, I recently debugged exactly that: GLFW only uses higher-level NS* objects, which are part of the Cocoa framework, and does not use any lower-level CGL* symbols. The OpenGL Framework is misleadingly named in that it contains the CGL implementation. The actual GL implementation is in `libGL.dylib`, which is also linked transitively through CoreVideo.

**Comment by jfirebaugh** [comment]
Ok, what does that mean for fixing the linker error?

**Comment by jfirebaugh** [comment]
@springmeyer Could you look into why the clang-3.8 build is failing?

**Comment by springmeyer** [comment]
> @springmeyer Could you look into why the clang-3.8 build is failing?

Yes. @jfirebaugh latest mason only provides clang++ stable releases of 3.8.1 and 3.9.1. So, I'd recommend upgrading to 3.9. This is available in #61

**Pull request merged**